frontend/packages/console-shared/src/hooks/useTableColumns.ts

```diff
@@ -0,0 +1,34 @@
+import * as React from 'react';
+import { useUserSettingsCompatibility } from './useUserSettingsCompatibility';
+import { COLUMN_MANAGEMENT_LOCAL_STORAGE_KEY } from '../constants';
+
+type TableColumnsType = {
+  [id: string]: string[];
+};
+
+export const COLUMN_MANAGEMENT_CONFIGMAP_KEY = 'console.tableColumns';
+
+export const useTableColumns = (
+  id: string,
+): [string[], (selectedColumns: Set<string>) => void, boolean] => {
+  const [tableColumns, setTableColumns, loaded] = useUserSettingsCompatibility<TableColumnsType>(
+    COLUMN_MANAGEMENT_CONFIGMAP_KEY,
+    COLUMN_MANAGEMENT_LOCAL_STORAGE_KEY,
+    {},
+    true,
+  );
+  const updateTableColumn = React.useCallback(
+    (selectedColumns: Set<string>) => {
+      setTableColumns((prevState) => {
+        return { ...prevState, [id]: [...selectedColumns] };
+      });
+    },
+    [id, setTableColumns],
+  );
+  const columns = React.useMemo(() => (loaded && tableColumns?.[id]) || [], [
+    loaded,
+    tableColumns,
+    id,
+  ]);
```

Comment:
You want to always maintain stable identity checks on data structures. If you don't you risk triggering dep changes of other hooks.
Of course this is short lived while `!loaded`.
I guess an alternative is to make `[]` a constant instead of using memo.